File: 0500_Keyboard_Row/solution.java
package cn.huang.leetcode;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/*
    500. Keyboard Row
 */
public class LC_0500_KeyboardRow {
    private static String rowOne = "qwertyuiopQWERTYUIOP";
    private static String rowTwo = "asdfghjklASDFGHJKL";
    private static String rowThree = "zxcvbnmZXCVBNM";

    public String[] findWords(String[] words) {
        int length = words.length;

        List<String> wordList = new ArrayList<String>();
        for(String word : words){
            if(canInput(word)){
                wordList.add(word);
            }
        }

        String[] results = new String[wordList.size()];
        return wordList.toArray(results);
    }
    private boolean canInput(String word){

        char c = word.charAt(0);
        String checkRow = rowThree;
        if(rowOne.indexOf(c) >= 0){
            checkRow = rowOne;
        }else if(rowTwo.indexOf(c) >= 0){
            checkRow = rowTwo;
        }

        for(int i = 1; i < word.length(); i++){
            char ch = word.charAt(i);
            if(checkRow.indexOf(ch) < 0){
                return false;
            }
        }

        return true;
    }
}

File: 0500_Keyboard_Row/solution.cpp
/*
500. Keyboard Row

Level: Easy

https://leetcode.com/problems/keyboard-row
*/
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>

#include <algorithm>

#include "common/ListNode.h"
#include "common/TreeNode.h"

using namespace std;

/*
  Solution: 
*/

class Solution
{
private:
  bool canInput(string word)
  {
    vector<string> rows = {
        "qwertyuiopQWERTYUIOP",
        "asdfghjklASDFGHJKL",
        "zxcvbnmZXCVBNM"};

    int row = 0;

    for (int i = 0; i < rows.size(); i++)
    {
      if (find(rows[i].begin(), rows[i].end(), word[0]) != rows[i].end())
      {
        row = i;
        break;
      }
    }

    for (char c : word)
    {
      if (rows[row].find(c) == std::string::npos)
      {
        return false;
      }
    }

    return true;
  }

public:
  vector<string> findWords(vector<string> &words)
  {
    vector<string> wordList;
    for (string word : words)
    {
      if (canInput(word))
      {
        wordList.push_back(word);
      }
    }

    return wordList;
  }
};